sky/backends/cloud_vm_backend.py

```diff
@@ -0,0 +1,351 @@
+
+    if not zones:
+        # For Azure, zones is always an empty list.
+        zone_str = 'all zones'
+    else:
+        zone_str = ','.join(z.name for z in zones)
+
+    if isinstance(cloud, clouds.Local):
+        logger.info(f'{style.BRIGHT}Launching on local cluster '
+                    f'{cluster_name!r}.')
+    else:
+        logger.info(f'{style.BRIGHT}Launching on {cloud} '
+                    f'{region_name}{style.RESET_ALL} ({zone_str})')
+    start = time.time()
+
+    # 5 seconds to 180 seconds. We need backoff for e.g., rate limit per
+    # minute errors.
+    backoff = common_utils.Backoff(initial_backoff=5,
+                                   max_backoff_factor=180 // 5)
+
+    # TODO(suquark): Should we just check the cluster status
+    #  if 'cluster_exists' is true? Then we can skip bootstrapping
+    #  etc if all nodes are ready. This is a known issue before.
+
+    try:
+        with backend_utils.safe_console_status(
+                f'[bold cyan]Bootstrapping configurations for '
+                f'[green]{cluster_name}[white] ...'):
+            config = provision.bootstrap(provider_name, region_name,
+                                         cluster_name, bootstrap_config)
+    except Exception:
+        logger.error('Failed to bootstrap configurations for '
+                     f'"{cluster_name}".')
+        raise
+
+    for retry_cnt in range(_MAX_RETRY):
+        try:
+            with backend_utils.safe_console_status(
+                    f'[bold cyan]Starting instances for '
+                    f'[green]{cluster_name}[white] ...'):
+                provision_metadata = provision.start_instances(provider_name,
+                                                               region_name,
+                                                               cluster_name,
+                                                               config=config)
+            break
+        except Exception as e:  # pylint: disable=broad-except
+            logger.debug(f'Starting instances for "{cluster_name}" '
+                         f'failed. Stacktrace:\n{traceback.format_exc()}')
+            if retry_cnt >= _MAX_RETRY - 1:
+                logger.error(f'Failed to provision "{cluster_name}" after '
+                             'maximum retries.')
+                raise e
+            sleep = backoff.current_backoff()
+            logger.info('Retrying launching in {:.1f} seconds.'.format(sleep))
+            time.sleep(sleep)
+
+    logger.debug(f'Waiting "{cluster_name}" to be started...')
+    with backend_utils.safe_console_status(
+            f'[bold cyan]Waiting '
+            f'[green]{cluster_name}[bold cyan] to be started...'):
+        # AWS would take a very short time (<<1s) updating the state of
+        # the instance. Wait 3 seconds should be enough.
+        time.sleep(3)
+        provision.wait_instances(provider_name, region_name, cluster_name,
+                                 'running')
```

Comment:
Do we need retry for this?

Reply:
I think we don't need retry any more with the sleep